### ISSUE TITLE
Feedback on the approved link error message

### ISSUE THREAD

**Issue by carolinan** [WordPress/theme-review-action#53]
The error message that displayed was:
```
"nayrathemes.com" found when viewing "/?p=1241" (via: single.php) is not an approved link.
See: https://github.com/WordPress/theme-review-action/blob/trunk/docs/ui-errors.md#page-should-not-have-unexpected-links
```

The theme author wrote: _We are not understand what need to do for theme review action._

https://themes.trac.wordpress.org/ticket/103019#comment:4

The documentation says:
```
Page should not have unexpected links
This test expects to not find links that are not approved.

Troubleshooting
Verify that the theme only includes links in the hosts list.
```

I don't think it is clear enough what a not approved or unexpected link is.
The hosts list is too difficult to interpret because it is not obvious that the theme and author URI are included on the list.

**Comment by kafleg** [comment]
Yes, I am also having a hard time understanding this notice. I think we need better text and examples to fix it.

**Comment by carolinan** [comment]
I think this test is too restrictive. 
It is an attempt to use an allowed list for links, but I think it will be too much work for us to maintain an allowed list.

I don't know how to allow front facing links, _but at the same time_ limit the footer credit link to theme and author URI.
_Enforce all external links in the front of the theme to include nofollow?_

If we remove both the test and the link requirements, this opens up for themes to be submitted for backlinks only.
For example, themes submitted only to promote unrelated products. And I mean unrelated like your uncle's car wash, not your premium plugin.
Do we consider that a problem or not? How can we give authors this freedom to add links and still prevent spam?

**Comment by jonoalderson** [comment]
These kinds of 'credit' footer links (and similar;  whether they're to website creators, hosting companies, or theme producers, etc) should always be nofollow'd as they're a form of 'paid advert'.

However, there are definitely legitimate cases for external footer links _not_ to be nofollow'd (a link to the parent company, or to a relevant resource, etc).

Difficult!
